drizzlepac/haputils/quality_analysis.py

```diff
@@ -425,9 +446,257 @@ def generate_plots(json_data):
+    x = plot_dict['x']
+    y = plot_dict['y']
+    title = plot_dict['title']
+    x_label = plot_dict['x_label']
+    y_label = plot_dict['y_label']
+    tips = plot_dict['tips']
+    source = plot_dict['source']
+    
+    # check for optional elements
+    color = plot_dict.get('color', 'blue')
+    click_policy = plot_dict.get('click_policy', 'hide')
+    markersize = plot_dict.get('markersize', 10)
+    colormap = plot_dict.get('colormap', False)
+    legend_group = plot_dict.get('legend_group')
+    
+    # Define a figure object
+    p1 = figure()
+
+    if colormap:
+        # Add the glyphs
+        p1.circle(x=x, y=y, source=source, 
+                  fill_alpha=0.5, line_alpha=0.5, 
+                  size=markersize, color='colormap',
+                  legend_group=legend_group)
+    else:
+        # Add the glyphs
+        p1.circle(x=x, y=y, source=source, 
+                  fill_alpha=0.5, line_alpha=0.5, 
+                  size=markersize)
 
+    p1.legend.click_policy = click_policy
     
+    p1.title.text = title
+    p1.xaxis.axis_label = x_label
+    p1.yaxis.axis_label = y_label
+
+    hover_p1 = HoverTool()
+    tools = build_tooltips(tips)
+    hover_p1.tooltips = tools
+    p1.add_tools(hover_p1)
     
+    return p1
     
+def generate_summary_plots(fit_data, output='cal_qa_results.html'):
+    """Generate the graphics associated with this particular type of data.
+
+    Parameters
+    ==========
+    fit_data: Pandas dataframe
+        Dataframe consisting of the relative alignment astrometric fit results
+    
+    NOTES
+    -----
+    Example from bokeh.org on how to create a tabbed set of plots:
+    
+    .. code:: python
+    
+        from bokeh.io import output_file, show
+        from bokeh.models import Panel, Tabs
+        from bokeh.plotting import figure
+
+        output_file("slider.html")
+
+        p1 = figure(plot_width=300, plot_height=300)
+        p1.circle([1, 2, 3, 4, 5], [6, 7, 2, 4, 5], size=20, color="navy", alpha=0.5)
+        tab1 = Panel(child=p1, title="circle")
+
+        p2 = figure(plot_width=300, plot_height=300)
+        p2.line([1, 2, 3, 4, 5], [6, 7, 2, 4, 5], line_width=3, color="navy", alpha=0.5)
+        tab2 = Panel(child=p2, title="line")
+
+        tabs = Tabs(tabs=[ tab1, tab2 ])
+
+        show(tabs)
+
+    
+    """
+    # TODO: include the date from the input data as part of the html filename
+    # Set the output file immediately as advised by Bokeh.
+    output_file(output)
+
+    # Setup the source of the data to be plotted so the axis variables can be
+    # referenced by column name in the Pandas dataframe
+    fitDF = ColumnDataSource(fit_data)
+    num_of_datasets = len(fit_data.index)
+    print('Number of datasets: {}'.format(num_of_datasets))
+    
+    colormap = [DETECTOR_LEGEND[x] for x in fitDF.data[HOVER_COLUMNS[1]]]
+    fitDF.data['colormap'] = colormap
+    inst_det = ["{}/{}".format(i,d) for (i,d) in zip(fitDF.data[HOVER_COLUMNS[0]], 
```

Comment:
OK, I am being picky mostly because I have trouble keeping track in Python of the type of objects being shuffled around.  fitDF here is a ColumnDataSource and not a dataframe.  I have to fix my code too!!!

Reply:
The variable 'fitDF' was also renamed to 'fitCDS' in 7ba7497.